bitcoin/BRSyncManager.c

```diff
@@ -2041,8 +2087,14 @@ static BRAddress *
 _getWalletAddresses (BRWallet *wallet,size_t *addressCount) {
     assert (addressCount);
 
+    // For BTC, `BRWalletAllAddrs() produces an array of SEGWIT addresses; for BCH, it
+    // produces an array of LEGACY addresses.  See `BRAddressFromHash160()`
     size_t addrCount = BRWalletAllAddrs (wallet, NULL, 0);
 
+    // We'll double up the number of addresses.  For BTC: SEGWIT + LEGACY; for BCH: LEGACY +
```

Comment:
Is it too much trouble to propagate if it is BCH/BTC to this point?

Reply:
I thought about it but biased towards minimal changes - once I realized BRSet would handle it.  Better to just pass an argument through?

Reply:
Maybe better to handle it a different way...  Don't add the LEGACY address here; instead when converting to string, for BTC, double up with SEGWIT + LEGACY? That then leaves BRWalletAllAddrs() as the set of addresses and leaves the string conversion for BTC and BCH in one place.

Reply:
I like that.